railgun/test/integration/knative_test.go

```diff
@@ -0,0 +1,257 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"time"
+
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	types "k8s.io/apimachinery/pkg/types"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/kind"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	knservingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	knativeversioned "knative.dev/serving/pkg/client/clientset/versioned"
+)
+
+const (
+	knativeCrds = "https://github.com/knative/serving/releases/download/v0.13.0/serving-crds.yaml"
+	knativeCore = "https://github.com/knative/serving/releases/download/v0.13.0/serving-core.yaml"
+)
+
+func isKnativeReady(ctx context.Context, cluster kind.Cluster, t *testing.T) bool {
+	return assert.Eventually(t, func() bool {
+		podList, err := cluster.Client().CoreV1().Pods("knative-serving").List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Logf("failed retrieving knative pods. %v", err)
+			return false
+		}
+
+		if len(podList.Items) != 4 {
+			t.Logf("expected 4 pods, found %d", len(podList.Items))
+			return false
+		}
+
+		for _, pod := range podList.Items {
+			if pod.Status.Phase != v1.PodRunning {
+				return false
+			}
+		}
+
+		t.Logf("All knative pods are up and ready.")
+		return true
+
+	}, 60*time.Second, 1*time.Second, true)
+}
+
+func TestKnativeIngress(t *testing.T) {
+	_ = proxyReady()
```

Comment:
Instead of using `retrieveProxyInfo`, the URLs for the proxy Admin API and the Proxy URL itself are available in the return here, we should use these instead.

```suggestion
	p = proxyReady()
```